src/sql/base/browser/ui/table/plugins/rowDetailView.ts

```diff
@@ -382,15 +387,30 @@ export class RowDetailView<T extends Slick.SlickData> {
 			//sneaky extra </div> inserted here-----------------v
 			html.push('<div class="detailView-toggle collapse"></div></div>');
 
-			html.push(`<div id='cellDetailView_${dataContext.id}' class='dynamic-cell-detail' `);   //apply custom css to detail
+			html.push(`<div id='cellDetailView_${this.getDataContextId(dataContext)}' class='dynamic-cell-detail' `);   //apply custom css to detail
 			html.push(`style=\'height:${dataContext._height}px;`); //set total height of padding
 			html.push(`top:${rowHeight}px'>`);             //shift detail below 1st row
-			html.push(`<div id='detailViewContainer_${dataContext.id}"'  class='detail-container' style='max-height:${(dataContext._height! - rowHeight + bottomMargin)}px'>`); //sub ctr for custom styling
-			html.push(`<div id='innerDetailView_${dataContext.id}'>${escape(dataContext._detailContent!)}</div></div>`);
+			html.push(`<div id='detailViewContainer_${this.getDataContextId(dataContext)}"'  class='detail-container' style='max-height:${(dataContext._height! - rowHeight + bottomMargin)}px'>`); //sub ctr for custom styling
+			let detailsContent = this._options.detailsHtml
+				? dataContext._detailContent!
```

Comment:
I'm not escaping to allow passing html here.  
So, HTML that is passed here in my case looks like `<span class='grid-cell-value-container'>escaped description</span> <a href='help link url'>Learn More</a>`

Reply:
I have added parameter "detailsHtml". If it is not passed, then content is escaped as it used to.
let detailsContent = this._options.detailsHtml ? dataContext._detailContent! : escape(dataContext._detailContent!);

Reply:
We can't allow unescaped HTML to come from an extension or any other uncontrolled source. That's highly dangerous from a security perspective since it would allow users to directly interact and modify things in the main process which could lead to attacks which may affect other parts of the application like queries.

I would suggest you remove this part for now and then when this gets in submit a separate PR so we can look at that specifically - this PR is too big and has too many other changes already.